File: GO/Peces.go
package main

// Interface para los métodos abstractos
type Peces interface {
	Escamas() string
}

// Struct “base” que simula la clase padre
type BasePeces struct {
	Nombre string
	Aletas int
	Branquias int 
}

// Constructor para BasePeces
func NewBasePeces(nombre string, aletas, branquias int) *BasePeces {
	return &BasePeces{
		Nombre: nombre, 
		Aletas: aletas,
		Branquias: branquias,
	}
}

File: C#/C#/clases_abstractas/Peces.cs
using System;

namespace C_.clases_abstractas
{
    internal abstract class Peces
    {
        public String nombre;
        public byte aletas; // Cantidad
        public byte branquias; // Cantidad

        // Constructor con parametros
        public Peces(String nombre, byte aletas, byte branquias)
        {
            this.nombre = nombre;
            this.aletas = aletas;
            this.branquias = branquias;
        }

        // Metodo abstracto
        public abstract void escamas();
    }
}
